### PR TITLE
Move sbom to internal, and add standard output tests

### PR DESCRIPTION
**Pull request by another-rex**
- Move SBOM versions enum into models folder. 
  - This change is done to proactively avoid future circular dependency issues
- Move CycloneDX logic into internal/output/sbom
  - Avoid making a public library yet for something that we might wish to change in the near future. 
- By having CycloneDX in output with the rest of the output formats, we can add the standard output tests as well.

### CODE REVIEW

**Comment by another-rex** [comment]
Depending on the order #1093 is merged, either this PR or #1093 needs to be updated.

File: internal/output/cyclonedx.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	"io"
+
+	"github.com/CycloneDX/cyclonedx-go"
+	"github.com/google/osv-scanner/internal/output/sbom"
+	"github.com/google/osv-scanner/internal/utility/purl"
+	"github.com/google/osv-scanner/pkg/models"
+)
+
+// PrintJSONResults writes results to the provided writer in JSON format
+func PrintCycloneDXResults(vulnResult *models.VulnerabilityResults, cycloneDXVersion models.CycloneDXVersion, outputWriter io.Writer) error {
```
Comment (G-Rath) [inline internal/output/cyclonedx.go:14]:
```suggestion
// PrintCycloneDXResults writes results to the provided writer in CycloneDX format
func PrintCycloneDXResults(vulnResult *models.VulnerabilityResults, cycloneDXVersion models.CycloneDXVersion, outputWriter io.Writer) error {
```

File: pkg/models/cyclonedx.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package models
+
+const (
+	CycloneDXVersion14 CycloneDXVersion = iota
+	CycloneDXVersion15
+)
+
+type CycloneDXVersion int
```
Comment (michaelkedar) [inline pkg/models/cyclonedx.go:8]: nit: can you move the `type` definition above the `const`?

**Review by G-Rath — APPROVED** [review]

**Review by michaelkedar — APPROVED** [review]

**Review by cuixq — APPROVED** [review]
LGTM - maybe explain a bit more in the description why we make this change?

**Pull request merged**